src/backend/renderer/gles/mod.rs

```diff
@@ -1385,20 +1226,38 @@ impl Gles2Renderer {
     }
 }
 
-impl ExportMem for Gles2Renderer {
-    type TextureMapping = Gles2Mapping;
+impl ExportMem for GlesRenderer {
+    type TextureMapping = GlesMapping;
 
     #[instrument(level = "trace", parent = &self.span, skip(self))]
     fn copy_framebuffer(
         &mut self,
         region: Rectangle<i32, BufferCoord>,
     ) -> Result<Self::TextureMapping, Self::Error> {
-        self.make_current()?;
+        if let Some(target) = self.target.as_ref() {
+            target.make_current_no_shadow(&self.gl, &self.egl, None)?;
+        } else {
+            unsafe {
+                self.egl.make_current()?;
+            }
+        }
+
+        let (internal, has_alpha) = self
+            .target
+            .as_ref()
+            .ok_or(GlesError::UnknownPixelFormat)?
+            .format()
+            .ok_or(GlesError::UnknownPixelFormat)?;
+        let (format, layout) = gl_read_for_internal(internal).ok_or(GlesError::UnknownPixelFormat)?;
+
         let mut pbo = 0;
         unsafe {
             self.gl.GenBuffers(1, &mut pbo);
             self.gl.BindBuffer(ffi::PIXEL_PACK_BUFFER, pbo);
-            let size = (region.size.w * region.size.h * 4) as isize;
+            let size = (region.size.w
+                * region.size.h
+                * (gl_bpp(format, layout).ok_or(GlesError::UnsupportedPixelLayout)? / 8) as i32)
```

Comment:
I find that a bit hard to read with the inline `ok_or` part, maybe something like `let bpp = gl_bpp(format, layout).ok_or(GlesError::UnsupportedPixelLayout)? / 8) as i32` would increase readability.